src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs

```diff
@@ -1757,9 +1849,22 @@ private string AddQueryParameter(object value)
 
         public string ParameterPrefix { get; set; } = DefaultParameterPrefix;
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void IsCurrentlySupported(string methodName)
```

Comment:
Method that starts with `Is...` suggests it's gonna return `true / false` not that it's gonna throw. Better name would be `AssertMethodIsCurrentlySupported()`

Reply:
done